File: rust/src/models/capacity_threshold.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CapacityThreshold {
    pub included_amount: i32,

    pub per_unit_overage: rust_decimal::Decimal,

    pub price: rust_decimal::Decimal,
}

impl CapacityThreshold {
    pub fn new(
        included_amount: i32,
        per_unit_overage: rust_decimal::Decimal,
        price: rust_decimal::Decimal,
    ) -> Self {
        Self {
            included_amount,
            per_unit_overage,
            price,
        }
    }
}

File: typescript/src/models/capacityThreshold.ts
// this file is @generated

export interface CapacityThreshold {
  includedAmount: number;

  perUnitOverage: string;

  price: string;
}

export const CapacityThresholdSerializer = {
  _fromJsonObject(object: any): CapacityThreshold {
    return {
      includedAmount: object["included_amount"],
      perUnitOverage: object["per_unit_overage"],
      price: object["price"],
    };
  },

  _toJsonObject(self: CapacityThreshold): any {
    return {
      included_amount: self.includedAmount,
      per_unit_overage: self.perUnitOverage,
      price: self.price,
    };
  },
};
